vllm/v1/request.py

```diff
@@ -57,6 +61,13 @@ def __init__(
         else:
             self.mm_inputs: List[MultiModalKwargs] = []
 
+        self.mm_hashes: Optional[List[
+            Optional[str]]] = self.inputs.multi_modal_hashes
```

Comment:
Why is this type annotated as `Optional[List[...]]` instead of `List[...]`?

Reply:
Because now we need to enable mm hash preprocessor to make the hash available (see PR description). I hope to remove this requirement before merging this PR.

Reply:
I think we should make this `List[Optional[str]]` actually because the default return value of `self.inputs.multi_modal_hashes` is `[]`